src/test/java/makaroshyna/onlinebookstore/service/BookServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+class BookServiceImplTest {
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
+
+    private BookService bookService;
+
+    @BeforeEach
+    void setUp() {
+        bookService = new BookServiceImpl(bookRepository, bookMapper, bookSpecificationBuilder);
+    }
+
+    @Test
+    @DisplayName("Verify save() method works")
+    public void save_ValidCreateBookRequestDto_ReturnsBookDto() {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto();
+        requestDto.setTitle("Test Title");
+
+        Book book = new Book();
+        book.setTitle("Test Title");
+
+        BookDto bookDto = new BookDto();
+        bookDto.setTitle("Test Title");
+        bookDto.setCategoryIds(Set.of());
+
+        when(bookMapper.toModel(requestDto)).thenReturn(book);
+        when(bookRepository.save(book)).thenReturn(book);
+        when(bookMapper.toDto(book)).thenReturn(bookDto);
+
+        BookDto result = bookService.save(requestDto);
+
+        assertNotNull(result);
+        assertEquals(result.getTitle(), requestDto.getTitle());
+    }
+
+    @Test
+    @DisplayName("Verify getAll() method works")
+    public void getAll_ValidPageable_ReturnsBookDto() {
+        when(bookRepository.findAll(any(Pageable.class)))
+                .thenReturn(new PageImpl<>(List.of(new Book())));
+        BookDto bookDto = new BookDto();
+        bookDto.setTitle("Test Title");
+        when(bookMapper.toDto(any())).thenReturn(bookDto);
+
+        List<BookDto> result = bookService.getAll(Pageable.unpaged());
+        assertNotNull(result);
+        assertEquals(result.size(), 1);
+    }
+
+    @Test
+    @DisplayName("Verify getById() method works")
+    public void getById_ValidId_ReturnsBookDto() {
+        Book book = new Book();
+        book.setId(1L);
+        book.setTitle("Test Title");
+
+        BookDto bookDto = new BookDto();
+        bookDto.setId(1L);
+        bookDto.setTitle("Test Title");
+
+        when(bookRepository.findById(1L)).thenReturn(Optional.of(book));
+        when(bookMapper.toDto(book)).thenReturn(bookDto);
+
+        BookDto result = bookService.getById(1L);
+        assertNotNull(result);
+        assertEquals(result.getTitle(), bookDto.getTitle());
+    }
+
+    @Test
+    @DisplayName("Verify updateById() method works")
+    public void updateById_ValidData_ReturnsBookDto() {
+        when(bookRepository.existsById(1L)).thenReturn(true);
+
+        CreateBookRequestDto requestDto = new CreateBookRequestDto();
+        requestDto.setTitle("Test Title");
+
+        Book book = new Book();
+        book.setId(1L);
+        book.setTitle("Test Title");
+
+        when(bookMapper.toModel(requestDto)).thenReturn(book);
+
+        BookDto bookDto = new BookDto();
+        bookDto.setId(1L);
+        bookDto.setTitle("Test Title");
+
+        when(bookMapper.toDto(any())).thenReturn(bookDto);
```

Comment:
I think it is better to keep things grouped by actions instead of entity type, please check other similar places as well
```suggestion
        CreateBookRequestDto requestDto = new CreateBookRequestDto();
        requestDto.setTitle("Test Title");

        Book book = new Book();
        book.setId(1L);
        book.setTitle("Test Title");

        BookDto bookDto = new BookDto();
        bookDto.setId(1L);
        bookDto.setTitle("Test Title");

        when(bookRepository.existsById(1L)).thenReturn(true);
        when(bookMapper.toModel(requestDto)).thenReturn(book);
        when(bookMapper.toDto(any())).thenReturn(bookDto);
```